.github/release_process.md

```diff
@@ -0,0 +1,28 @@
+# Change Review and Release Process
+
+Before Merging a pull request:
+
+- [ ] Bump version number in [setup.py](../setup.py) and [package.json](../package.json)
+- [ ] Run `make javascript sass` if your changes include JS/CSS changes
+- [ ] Get a green Travis build for this PR
```

Comment:
what about adding  a step to: `make check_translations_up_to_date` and check in any files generated by that command. To remind us of the command to update translations?

Reply:
That's in the developer checklist right now in `pull_request_template`. I'm still unsure of which pieces I want to put where but everything should still be covered.

Reply:
@Andytr1, I updated the release process to have that script